src/Microsoft.Azure.WebJobs.Host/Lease/LeaseFactory.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Host.Lease;
+using Microsoft.Azure.WebJobs.Host.Storage;
+
+namespace Microsoft.Azure.WebJobs.Host.Lease
+{
+    /// <summary>
+    /// ILeaseProxy factory
+    /// </summary>
+    internal class LeaseFactory
+    {
+        public static ILeaseProxy CreateLeaseProxy(IStorageAccountProvider storageAccountProvider)
+        {
+            ILeaseProxy leaseProxy = null;
+
+            // Choose between Sql and Blob based lease implementations
+            if (SqlLeaseProxy.IsSqlLeaseType())
```

Comment:
In case of an invalid connection string, attempting to perform any operation on the `ILeaseProxy` interface would fail at a later point. I don't think we should be attempting to fallback to storage at this point.